app/controllers/api/vms_controller.rb

```diff
@@ -228,6 +228,16 @@ def request_console_resource(type, id = nil, data = nil)
       end
     end
 
+    def assign_tags_resource(type, id, data)
+      vm = resource_search(id, type, collection_class(:vms))
+      tag_results = data['tags'].collect do |tag|
+        tags_assign_resource(vm, type, tag['id'], tag)
+      end
+      { id => tag_results }
+    rescue => err
+      { id => action_result(false, err.to_s) }
+    end
```

Comment:
Looking at the api.yml, I see bulk assign_tags and unassign_tags could be implemented to the following collections for completion:

```
blueprints
clusters
data_stores
groups
hosts
providers
resource_pools
service_templates
services
templates
tenants
users
```

so generalizing assign_tags_resource and unassign_tags_resource, maybe as a mixin would be a 👍